### PR TITLE
Feature/nijo 440 including currency converter

### PR DESCRIPTION
**Pull request by mingazhev**
In this PR:
- Removed old CurrencyRate service and it's artifacts, replaced with new implementation
- Added currency conversion to show prices for 1,2,3 steps of availability search based on Customer preferred currency
- Refactored markup application and shortened replaced some weird models with generic

### CODE REVIEW

File: Api/Services/Accommodations/AvailabilityService.cs [hunk 1]
```diff
@@ -125,58 +140,70 @@ ProviderData<SingleAccommodationAvailabilityDetails> AddProviderData(SingleAccom
...
+            var (_, _, settings, _) = await _customerSettingsManager.GetUserSettings(customer);
+            var preferredCurrency = settings.PreferredCurrency;
+
+            if (preferredCurrency == default)
+                return details;
+
+            return await func(details, async (price, currency) =>
+            {
+                var newCurrency = preferredCurrency;
+                var newPrice = price * await _currencyRateService.Get(currency, preferredCurrency);
```
Comment (kirillta) [inline Api/Services/Accommodations/AvailabilityService.cs]: This conversion is pretty naive

Comment (mingazhev) [reply Api/Services/Accommodations/AvailabilityService.cs]: What do you mean?
It works incorrect, slowly?

Comment (mingazhev) [reply Api/Services/Accommodations/AvailabilityService.cs]: If it's about no error handling, you're right, fixing that now

Comment (kirillta) [reply Api/Services/Accommodations/AvailabilityService.cs]: Not only that — ceiling, sane checks and other.
Should we make a separate library for that?

**Pull request merged**